staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -43,7 +43,10 @@ const (
 	replaceDirective = "replace"
 	mergeDirective   = "merge"
 
+	replaceKeysStrategy = "replaceKeys"
+
 	deleteFromPrimitiveListDirectivePrefix = "$deleteFromPrimitiveList"
+	replaceKeysDirective                   = "$" + replaceKeysStrategy
```

Comment:
Ones starting with `$` is used as key in the map, the others without `$` are used as value in the map.
There are a lot of example in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch_test.go